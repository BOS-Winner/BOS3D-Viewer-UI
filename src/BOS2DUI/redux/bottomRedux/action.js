import * as actionType from "./actionType";

export function changeMode(mode) {
  return {
    type: actionType.CHANGE_MODE,
    mode,
  };
}

export function showSetting(show) {
  return {
    type: actionType.SHOW_SETTING,
    show,
  };
}

export function showCptInfo(show) {
  return {
    type: actionType.SHOW_CPTINFO,
    show,
  };
}

export function showLayerSwitcher(show) {
  return {
    type: actionType.SHOW_LAYER_SWITCHER,
    show,
  };
}

export function updateLayerList(layers) {
  return {
    type: actionType.UPDATE_LAYER_LIST,
    layers,
  };
}

/**
 * 打开或隐藏批注面板
 * @param {boolean} show - 是否显示
 * @return {{show: boolean, type: string}}
 */
export function showAnnotationList(show) {
  return {
    type: actionType.SHOW_ANNOTATION_LIST,
    show,
  };
}

/**
 * 打开或隐藏批注编辑器
 * @param {boolean} show - 是否显示
 * @param {object} [data] - 是否使用指定预设数据
 * @return {{show: boolean, type: string, data:object}}
 */
export function showAnnotationEditor(show, data) {
  return {
    type: actionType.SHOW_ANNOTATION_EDITOR,
    show,
    data,
  };
}
