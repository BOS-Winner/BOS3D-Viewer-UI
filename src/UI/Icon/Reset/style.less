@import "../../theme/index";

.reset {
  padding: 16px 8px;
  .item {
    box-sizing: border-box;
    background: #2E2F30;
    border-radius: 2px;
    border: 1px solid rgba(255, 255, 255, 0.1);
    padding: 6px;
    margin-bottom: 12px;
  }
  .line{
    margin: 17px 0;
    height: 1px;
    width: 100%;
    background:rgba(255, 255, 255, 0.08);
    
  }
  :global {
    .bos3d-checkbox-wrapper{
      color: #DDDDDD;
    }
    .bos3d-checkbox-inner{
      width: 14px;
      height: 14px;
      
    }
  }
}
