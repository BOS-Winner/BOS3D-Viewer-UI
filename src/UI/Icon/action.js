import * as actionType from "./actionType";

/**
 * 改变当前UI模式
 * @param {string} mode - 模式名称
 * @return {{mode: string, type: string}}
 */
export function changeMode(mode, exitMode = "") {
  return {
    type: actionType.CHANGE_MODE,
    mode,
    exitMode,
  };
}

/**
 * 更新操作列表
 * @param {object} obj - 操作数据，包括key和操作名称，其他必需数据
 * @param {string} obj.type - 操作类型（方法）: hide, wireframe, isolate, colorful
 * @param {string} obj.name - 名称
 * @param {string[]} obj.keys - 操作的构件key数组
 * @param {number} [obj.color] - 构件改变的颜色
 * @return {{item: object, type: string}}
 */
export function updateList(obj) {
  return {
    type: actionType.UPDATE_LIST,
    item: obj
  };
}

/**
 * 移动操作列表尾指针
 * @param {number} pointer - 新的指针位置
 * @return {{pointer: number, type: string}}
 */
export function movePointer(pointer) {
  return {
    type: actionType.MOVE_POINTER,
    pointer
  };
}

/**
 * 重设操作历史
 * @return {{type: string}}
 */
export function resetHistory() {
  return {
    type: actionType.RESET_HISTORY
  };
}

/**
 * 修改鼠标图标
 * @param mouseIcon
 * @return {{mouseIcon: *, type: string}}
 */
export function changeMouseIcon(mouseIcon) {
  return {
    type: actionType.CHANGE_MOUSE_ICON,
    mouseIcon,
  };
}

/**
 * 构件信息是否显示
 * @param visible
 * @return {{visible: boolean, type: string}}
 */
export function setComponentInfoVisible(visible) {
  return {
    type: actionType.COMPONENT_INFO_VISIBLE,
    visible
  };
}

/**
 * 部件族信息是否显示
 * @param {object} payload 部件族信息显示的参数
 */
export function setFamilyInfoVisible(payload) {
  return {
    type: actionType.FAMILY_INFO_VISIBLE,
    payload
  };
}
