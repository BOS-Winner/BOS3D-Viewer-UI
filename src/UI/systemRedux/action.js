import * as actionType from "./actionType";

/**
 * 保存模型信息到reducer
 */
export function updateModelDetail(payload) {
  return {
    type: actionType.UPDATE_MODEL_DETAIL,
    payload
  };
}
