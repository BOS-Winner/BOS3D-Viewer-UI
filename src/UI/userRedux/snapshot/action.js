import * as actionType from "./actionType";

export function add(snapshot) {
  return {
    type: actionType.ADD,
    snapshot
  };
}

export function remove(code) {
  return {
    type: actionType.REMOVE,
    code
  };
}

export function update(code, snapshot) {
  return {
    type: actionType.UPDATE,
    snapshot,
    code
  };
}

export function clear() {
  return {
    type: actionType.CLEAR
  };
}
