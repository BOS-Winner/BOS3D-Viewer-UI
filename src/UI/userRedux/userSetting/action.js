import * as actionType from "./actionType";

export function changeToolbarState(name, enable) {
  return {
    type: actionType.CHANGE_TOOLBAR_STATE,
    name,
    enable,
  };
}

export function setToolbarState(toolState) {
  return {
    type: actionType.SET_TOOLBAR_STATE,
    toolState,
  };
}

export function setCustomToolbarState(toolState) {
  return {
    type: actionType.SET_CUSTOMTOOLSTATE,
    customToolState: toolState,
  };
}

export function changeDisplaySetting(name, value) {
  return {
    type: actionType.CHANGE_DISPLAY_SETTING,
    name,
    value,
  };
}

export function changeCameraSetting(name, value) {
  return {
    type: actionType.CHANGE_CAMERA_SETTING,
    name,
    value
  };
}

export function restoreSetting(settingType) {
  return {
    type: actionType.RESTORE_SETTING,
    settingType,
  };
}

export function changeModelSetting(modelKey, name, value) {
  return {
    type: actionType.CHANGE_MODEL_SETTING,
    modelKey,
    name,
    value,
  };
}

export function setSetting(setting) {
  return {
    type: actionType.SET_SETTING,
    setting,
  };
}

export function changeBestView(visible = false) {
  return {
    type: actionType.CHANGE_BEST_VIEW,
    visible,
  };
}
