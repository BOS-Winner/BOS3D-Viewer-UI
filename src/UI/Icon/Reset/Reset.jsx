import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Checkbox } from 'antd';
import { resetHistory } from "../action";
import Modal from "../../Base/Modal";
import style from "./style.less";
import { DEFAULT_MODAL_PLACE } from "../../constant";

const OPTIONS_LIST = [
  { value: 'visible', label: '构件可见' },
  { value: 'highlight', label: '构件高亮' },
  { value: 'perspective', label: '模型视角' },
  { value: 'wireframe', label: '构件线框' },
  { value: 'colorful', label: '构件颜色' },
];

class Reset extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      checkedList: [],
      isCheckALL: false
    };
  }

  onChange = (checkedValues) => {
    const isCheckALL = checkedValues.length === OPTIONS_LIST.length;
    this.setState({
      checkedList: checkedValues,
      isCheckALL
    });
  }

  onCheckAll = (e) => {
    const checked = e.target.checked;
    let checkedList = [];
    if (checked) {
      checkedList = OPTIONS_LIST.map(item => item.value);
    }
    this.setState({
      isCheckALL: checked,
      checkedList
    });
  }

  onReset(e) {
    e.preventDefault();
    e.stopPropagation();
    const { checkedList } = this.state;
    const visible = checkedList.some(item => item === 'visible');
    const highlight = checkedList.some(item => item === 'highlight');
    const wireframe = checkedList.some(item => item === 'wireframe');
    const colorful = checkedList.some(item => item === 'colorful');
    const perspective = checkedList.some(item => item === 'perspective');
    this.props.viewer.resetScene({
      visible,
      selected: highlight,
      wireframed: wireframe,
      colorfully: colorful,
      view: perspective
    });
  }

  render() {
    const { isCheckALL, checkedList } = this.state;
    return (
      <Modal
        title="复位设置"
        visible={this.props.visible}
        onCancel={() => this.props.onHide()}
        viewportDiv={this.props.viewer.viewportDiv}
        top={DEFAULT_MODAL_PLACE.reset.top}
        right={DEFAULT_MODAL_PLACE.reset.right}
        left={DEFAULT_MODAL_PLACE.reset.left}
        width="350px"
        height="auto"
        minWidth={163}
        minHeight={216}
      >
        <div className={style.reset}>
          <Checkbox.Group style={{ width: '100%' }} value={checkedList} onChange={this.onChange}>
            {OPTIONS_LIST.map(item => (
              <div key={item.value} className={style.item}>
                <Checkbox value={item.value}>{item.label}</Checkbox>
              </div>
            ))}
          </Checkbox.Group>
          <div className={style.item}>
            <Checkbox checked={isCheckALL} onChange={this.onCheckAll}>全选</Checkbox>
          </div>
          <div className={style.line} />
          <button
            className="bos-btn bos-btn-primary bos-btn-block"
            type="submit"
            disabled={checkedList.length === 0}
            onClick={e => { this.onReset(e) }}
          >
            复位选中项目
          </button>
        </div>
      </Modal>
    );
  }
}

Reset.propTypes = {
  clearUndoList: PropTypes.func.isRequired,
  viewer: PropTypes.object.isRequired,
  visible: PropTypes.bool.isRequired,
  onHide: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  viewer: state.system.viewer3D,
});
const mapDispatchToProps = (dispatch) => ({
  clearUndoList: () => { dispatch(resetHistory()) }
});
const WrappedContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(Reset);

export default WrappedContainer;
