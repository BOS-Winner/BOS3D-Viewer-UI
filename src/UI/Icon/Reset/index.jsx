import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import Icon from "Base/Icon";
import Reset from "./Reset";
// import resetPng from "../img/white/reset.png";
import { AntdIcon } from '../../utils/utils';
import iconstyle from '../../Toolbar/bottom.less';

class ResetContainer extends React.Component {
  static propTypes = {
    mode: PropTypes.string.isRequired,
  }

  constructor(props) {
    super(props);
    this.state = {
      show: false,
    };
  }

  componentDidUpdate(prevProps) {
    if (prevProps.mode !== this.props.mode && this.props.mode === "漫游模式") {
      // eslint-disable-next-line react/no-did-update-set-state
      this.setState({
        show: false,
      });
    }
  }

  onClick(e) {
    e.preventDefault();
    e.stopPropagation();
    this.setState(state => ({
      show: !state.show
    }));
  }

  render() {
    return (
      <div
        title="复位"
        role="button"
        tabIndex={0}
        onClick={e => { this.onClick(e) }}
      >
        <Reset
          visible={this.state.show}
          onHide={() => this.setState({ show: false })}
        />
        <Icon
          icon={<AntdIcon type="iconreset" className={iconstyle.icon} />}
          title=""
          selected={this.state.show}
          showTitle={false}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  mode: state.button.mode,
});
const mapDispatchToProps = () => ({});
const WrappedContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(ResetContainer);
export default WrappedContainer;
