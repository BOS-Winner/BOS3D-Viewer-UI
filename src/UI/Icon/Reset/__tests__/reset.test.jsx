import React from 'react';
import { render, act, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { genStore } from 'jest/initTest';
import Reset from '../Reset';
import { updateList } from '../../action';

jest.mock('Base/Modal', () => {
  // eslint-disable-next-line global-require
  const Modal = require('mock/Modal').default;
  return props => <Modal {...props} />;
});
const store = genStore();
store.getState().system.viewer3D.resetScene = jest.fn();
const onHide = jest.fn();
let resetContainer;
let resetGetByText;
let resetRerender;

describe('reset', () => {
  beforeEach(() => {
    const { container, getByText, rerender } = render((
      <Provider store={store}>
        <Reset visible onHide={onHide} />
      </Provider>
    ));
    resetContainer = container;
    resetGetByText = getByText;
    resetRerender = rerender;
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('should show and hide by calling onHide', () => {
    expect(resetContainer).toMatchSnapshot();

    resetGetByText('close mock modal').click();
    expect(onHide).toBeCalledTimes(1);

    resetRerender((
      <Provider store={store}>
        <Reset visible={false} onHide={onHide} />
      </Provider>
    ));
    expect(resetContainer).toMatchSnapshot();
  });

  it('should check what you need', () => {
    const all = resetGetByText('全选').firstElementChild;
    const visible = resetGetByText('构件可见').firstElementChild;
    const highlight = resetGetByText('构件高亮').firstElementChild;
    const camera = resetGetByText('模型视角').firstElementChild;
    const wireframe = resetGetByText('构件线框').firstElementChild;
    const colorful = resetGetByText('构件颜色').firstElementChild;
    act(() => {
      fireEvent.input(all, {
        checked: true,
      });
    });
    expect(all.checked).toBe(true);
    expect(visible.checked).toBe(true);
    expect(highlight.checked).toBe(true);
    expect(camera.checked).toBe(true);
    expect(wireframe.checked).toBe(true);
    expect(colorful.checked).toBe(true);

    act(() => {
      fireEvent.input(all, {
        checked: false,
      });
    });
    expect(all.checked).toBe(false);
    expect(visible.checked).toBe(false);
    expect(highlight.checked).toBe(false);
    expect(camera.checked).toBe(false);
    expect(wireframe.checked).toBe(false);
    expect(colorful.checked).toBe(false);

    act(() => {
      fireEvent.input(visible, {
        checked: true,
      });
    });
    expect(visible.checked).toBe(true);
    act(() => {
      fireEvent.input(visible, {
        checked: false,
      });
    });
    expect(visible.checked).toBe(false);

    act(() => {
      fireEvent.input(highlight, {
        checked: true,
      });
    });
    expect(highlight.checked).toBe(true);
    act(() => {
      fireEvent.input(highlight, {
        checked: false,
      });
    });
    expect(highlight.checked).toBe(false);

    act(() => {
      fireEvent.input(camera, {
        checked: true,
      });
    });
    expect(camera.checked).toBe(true);
    act(() => {
      fireEvent.input(camera, {
        checked: false,
      });
    });
    expect(camera.checked).toBe(false);

    act(() => {
      fireEvent.input(wireframe, {
        checked: true,
      });
    });
    expect(wireframe.checked).toBe(true);
    act(() => {
      fireEvent.input(wireframe, {
        checked: false,
      });
    });
    expect(wireframe.checked).toBe(false);

    act(() => {
      fireEvent.input(colorful, {
        checked: true,
      });
    });
    expect(colorful.checked).toBe(true);
    act(() => {
      fireEvent.input(colorful, {
        checked: false,
      });
    });
    expect(colorful.checked).toBe(false);
  });

  it('should call resetScene', () => {
    const all = resetGetByText('全选').firstElementChild;
    store.dispatch(updateList({
      type: 'hide',
      name: '隐藏',
      keys: ['M123456']
    }));
    act(() => {
      fireEvent.input(all, {
        checked: true,
      });
    });
    act(() => {
      resetGetByText('复位选中项目').click();
    });
    expect(store.getState().system.viewer3D.resetScene).lastCalledWith({
      visible: true,
      selected: true,
      wireframed: true,
      colorfully: true,
      view: true,
    });
    expect(store.getState().button.history.length).toBe(0);
    expect(store.getState().button.pointer).toBe(0);
  });
});
