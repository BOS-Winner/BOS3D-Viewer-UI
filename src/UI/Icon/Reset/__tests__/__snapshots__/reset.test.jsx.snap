// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reset should show and hide by calling onHide 1`] = `
<div>
  <div
    style="display: block; position: absolute; bottom: 20%; left: 20px; width: 163px; height: 216px;"
    title="复位设置"
  >
    <div>
      <span>
        复位设置
      </span>
      <button
        data-testid="close-mock-modal"
        type="button"
      >
        close mock modal
      </button>
    </div>
    <form
      class="reset"
    >
      <label>
        <input
          name="visible"
          type="checkbox"
        />
        构件可见
      </label>
      <label>
        <input
          name="highlight"
          type="checkbox"
        />
        构件高亮
      </label>
      <label>
        <input
          name="perspective"
          type="checkbox"
        />
        模型视角
      </label>
      <label>
        <input
          name="wireframe"
          type="checkbox"
        />
        构件线框
      </label>
      <label>
        <input
          name="colorful"
          type="checkbox"
        />
        构件颜色
      </label>
      <label>
        <input
          name="all"
          type="checkbox"
        />
        全选
      </label>
      <button
        type="submit"
      >
        复位选中项目
      </button>
    </form>
  </div>
</div>
`;

exports[`reset should show and hide by calling onHide 2`] = `
<div>
  <div
    style="display: none; position: absolute; bottom: 20%; left: 20px; width: 163px; height: 216px;"
    title="复位设置"
  >
    <div>
      <span>
        复位设置
      </span>
      <button
        data-testid="close-mock-modal"
        type="button"
      >
        close mock modal
      </button>
    </div>
    <form
      class="reset"
    >
      <label>
        <input
          name="visible"
          type="checkbox"
        />
        构件可见
      </label>
      <label>
        <input
          name="highlight"
          type="checkbox"
        />
        构件高亮
      </label>
      <label>
        <input
          name="perspective"
          type="checkbox"
        />
        模型视角
      </label>
      <label>
        <input
          name="wireframe"
          type="checkbox"
        />
        构件线框
      </label>
      <label>
        <input
          name="colorful"
          type="checkbox"
        />
        构件颜色
      </label>
      <label>
        <input
          name="all"
          type="checkbox"
        />
        全选
      </label>
      <button
        type="submit"
      >
        复位选中项目
      </button>
    </form>
  </div>
</div>
`;
